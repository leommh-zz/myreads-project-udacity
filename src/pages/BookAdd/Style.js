const style = {
    main: {
        backgroundColor: '#605872',
        flex: 1,
        padding: 20,
    },
}

export default style;
