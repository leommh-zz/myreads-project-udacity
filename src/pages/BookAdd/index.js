import React, { Component } from 'react';
import { Grid } from '@material-ui/core';
import SearchInput from '../../components/SearchInput';
import ButtonBack from '../../components/ButtonBack';
import BookCase from '../../components/BookCase';
import Style from './Style';

class BookAdd extends Component{
    /**
     * @constructor 
     * @description - Clean screen search!
     */
    constructor(props){
        super(props);
        const { search } = props;
        search('');
    }

    render(){
        const { defineStatus, loading, title, booksSearch, search } = this.props;
        return (
            <Grid 
            container 
            style={Style.main}  
            direction="column" 
            justify="center"
            alignItems="center" 
        >
            <Grid item>
                <ButtonBack />
            </Grid>
            <Grid item>
                <SearchInput search={search}  />
            </Grid>
            <Grid item>
                <BookCase title={title} books={booksSearch} defineStatus={defineStatus} loading={loading} /> 
            </Grid>
        
        </Grid>
        );
    }

} 

export default BookAdd;