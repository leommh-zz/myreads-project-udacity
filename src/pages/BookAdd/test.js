import React from 'react';
import { shallow } from 'enzyme';
import BookAdd from './index';

describe('[Component] BookAdd', () => {
    const booksSearch = [
        {
            title: "Title Book",
            description: "This book and the table",
            categories: [
                "action"
            ],
            authors: [
                "Leonardo Morini"
            ],
            shelf: '',
            averageRating: 2
        }
    ]

    const search = () => console.log('Searched');
    const defineStatus = () => console.log('Updated');
    const loading = false;

    const wrapper = shallow(
        <BookAdd 
            title='test' 
            booksSearch={booksSearch} 
            search={search} 
            loading={loading} 
            defineStatus={defineStatus} 
        />
    );

    it('Display Book Component with props', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
});