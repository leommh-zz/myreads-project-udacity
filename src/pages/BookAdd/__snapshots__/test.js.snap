// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] BookAdd Display Book Component with props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookAdd
    booksSearch={
        Array [
            Object {
              "authors": Array [
                "Leonardo Morini",
              ],
              "averageRating": 2,
              "categories": Array [
                "action",
              ],
              "description": "This book and the table",
              "shelf": "",
              "title": "Title Book",
            },
          ]
    }
    defineStatus={[Function]}
    loading={false}
    search={[Function]}
    title="test"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": Array [
        <WithStyles(Grid)
          item={true}
>
          <ButtonBack />
</WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
>
          <SearchInput
                    search={[Function]}
          />
</WithStyles(Grid)>,
        <WithStyles(Grid)
          item={true}
>
          <BookCase
                    books={
                              Array [
                                        Object {
                                          "authors": Array [
                                            "Leonardo Morini",
                                          ],
                                          "averageRating": 2,
                                          "categories": Array [
                                            "action",
                                          ],
                                          "description": "This book and the table",
                                          "shelf": "",
                                          "title": "Title Book",
                                        },
                                      ]
                    }
                    defineStatus={[Function]}
                    loading={false}
                    title="test"
          />
</WithStyles(Grid)>,
      ],
      "container": true,
      "direction": "column",
      "justify": "center",
      "style": Object {
        "backgroundColor": "#605872",
        "flex": 1,
        "padding": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ButtonBack />,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <SearchInput
            search={[Function]}
/>,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "search": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <BookCase
            books={
                        Array [
                                    Object {
                                      "authors": Array [
                                        "Leonardo Morini",
                                      ],
                                      "averageRating": 2,
                                      "categories": Array [
                                        "action",
                                      ],
                                      "description": "This book and the table",
                                      "shelf": "",
                                      "title": "Title Book",
                                    },
                                  ]
            }
            defineStatus={[Function]}
            loading={false}
            title="test"
/>,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "books": Array [
              Object {
                "authors": Array [
                  "Leonardo Morini",
                ],
                "averageRating": 2,
                "categories": Array [
                  "action",
                ],
                "description": "This book and the table",
                "shelf": "",
                "title": "Title Book",
              },
            ],
            "defineStatus": [Function],
            "loading": false,
            "title": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": Array [
          <WithStyles(Grid)
            item={true}
>
            <ButtonBack />
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
>
            <SearchInput
                        search={[Function]}
            />
</WithStyles(Grid)>,
          <WithStyles(Grid)
            item={true}
>
            <BookCase
                        books={
                                    Array [
                                                Object {
                                                  "authors": Array [
                                                    "Leonardo Morini",
                                                  ],
                                                  "averageRating": 2,
                                                  "categories": Array [
                                                    "action",
                                                  ],
                                                  "description": "This book and the table",
                                                  "shelf": "",
                                                  "title": "Title Book",
                                                },
                                              ]
                        }
                        defineStatus={[Function]}
                        loading={false}
                        title="test"
            />
</WithStyles(Grid)>,
        ],
        "container": true,
        "direction": "column",
        "justify": "center",
        "style": Object {
          "backgroundColor": "#605872",
          "flex": 1,
          "padding": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ButtonBack />,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <SearchInput
              search={[Function]}
/>,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "search": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <BookCase
              books={
                            Array [
                                          Object {
                                            "authors": Array [
                                              "Leonardo Morini",
                                            ],
                                            "averageRating": 2,
                                            "categories": Array [
                                              "action",
                                            ],
                                            "description": "This book and the table",
                                            "shelf": "",
                                            "title": "Title Book",
                                          },
                                        ]
              }
              defineStatus={[Function]}
              loading={false}
              title="test"
/>,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "books": Array [
                Object {
                  "authors": Array [
                    "Leonardo Morini",
                  ],
                  "averageRating": 2,
                  "categories": Array [
                    "action",
                  ],
                  "description": "This book and the table",
                  "shelf": "",
                  "title": "Title Book",
                },
              ],
              "defineStatus": [Function],
              "loading": false,
              "title": "test",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
